### PR TITLE
Preliminary label documentation

### PR DESCRIPTION
**Pull request by mgaudet**
Starts trying to address Issue #1953

Signed-off-by: Matthew Gaudet <magaudet@ca.ibm.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,12 +28,36 @@ We welcome and enourage all kinds of contribution to the project, not only code.
...
+* **`help wanted`** issues have value to the project but no immediate
+  human resources to undertake it. Those that are looking to complete a task
+  that someone isn't already working on can consider these.
+
+* **`bug`** issues are functional problems, errors or unexpected behaviour.
+
+* **`epic`** issues are used to group together related issues and to track larger 
+  goals in the project across issues. 
+
+
```
Comment (0xdaryl) [inline CONTRIBUTING.md]: * **`ci`** labels are used for issues and enhancements with the continuous integration system for pull request testing (e.g., Jenkins, Travis, AppVeyor, etc.)

* **`documentation`** labels are used for issues or enhancements to the documentation (either in the source code itself or stand-alone documentation files).

* **`license`** labels are used to annotate issues concerning the source code license.

Labels are also used to classify issues and pull requests by the dominant Eclipse OMR technology component they affect.  For instance:
* **`compiler`**
* **`gc`**
* **`glue`**
* **`jitbuilder`**
* **`port`**
* **`ras`**
* **`thread`**
* **`tril`**

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,12 +28,36 @@ We welcome and enourage all kinds of contribution to the project, not only code.
...
+  with part of the code base.
+
+* **`help wanted`** issues have value to the project but no immediate
+  human resources to undertake it. Those that are looking to complete a task
+  that someone isn't already working on can consider these.
+
+* **`bug`** issues are functional problems, errors or unexpected behaviour.
+
+* **`epic`** issues are used to group together related issues and to track larger 
+  goals in the project across issues.
```
Comment (0xdaryl) [inline CONTRIBUTING.md]: Perhaps list these alphabetically.

File: CONTRIBUTING.md (hunk 1 above)
Comment (0xdaryl) [reply CONTRIBUTING.md]: * **`build/configure`** labels are used to describe issues with the build and configure system (e.g., makefiles, autotools configure).
* **`cmake`** labels are similar to `build/configure` but they apply specifically to the Cmake configuration.
* **`test`** labels are for issues relating to the unit test framework and the unit tests themselves found in the `fvtest` directory.
* **`GSoC Project`** labels are for potential ideas for Google Summer Of Code projects.
* **`tooling`** labels are for issues concerning support tooling needed to support any of the code or processes within the project.

**Comment by mgaudet** [comment]
Thanks @0xdaryl I've added your label descriptions.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -28,12 +28,74 @@ We welcome and enourage all kinds of contribution to the project, not only code.
...
+
+Our GitHub issues get labelled by committers in order to allow easier navigation,
+and to flag certain issues as being of interest to certain groups. A subset of the 
+labels are documented below.
+
+* **`beginner`** generally refers to a task that would be suitable for someone new
+  to the project with little experience with either the technology or even
+  open-source projects and processes. They are intended for anyone who wants to
+  gently get their feet wet on building, testing, or familiarizing themselves
+  with part of the code base.
```
Comment (0xdaryl) [inline CONTRIBUTING.md]: You might also want to say:
"Multiple labels may be associated with an issue or PR: as many as needed to provide adequate categorization."

Comment (mgaudet) [reply CONTRIBUTING.md]: Done.

**Comment by rwy0717** [comment]
You could get fancy, and for each label, provide a link to the labelled issues 🕺 .

**Comment by mgaudet** [comment]
Links added.

**Comment by mstoodle** [comment]
lgtm @charliegracie any concerns ?

**Comment by 0xdaryl** [comment]
As all project leads have approved, I will merge.

**Pull request merged**